src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/whenshouldbeused/InputWhenShouldBeUsedSwitchExpression.java

```diff
@@ -0,0 +1,108 @@
+/*
+WhenShouldBeUsed
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.whenshouldbeused;
+
+public class InputWhenShouldBeUsedSwitchExpression {
+
+    void test(Object j, int x) {
+        int a = switch (j) {
+            case Point(int _, int y): // violation, 'When should be used instead of a single if*.'
+                if (y == 0) {
+                    yield 0;
+                }
+            case Integer _: if (x == 0)  yield 4;
```

Comment:
Can we have an example of a violation where case has a `{}` surrounding it?

Reply:
done